integration-tests/src/evm.rs

```diff
@@ -1840,6 +1851,75 @@ mod chainlink_precompile {
 			);
 		});
 	}
+
+	fn prices_should_be_comparable_with_dia(asset: AssetId, reference_oracle: EvmAddress) {
+		TestNet::reset();
+		hydra_live_ext("evm-snapshot/router").execute_with(|| {
+			let base = 10;
+			let route = vec![
+				Trade {
+					pool: PoolType::Omnipool,
+					asset_in: asset,
+					asset_out: 102,
+				},
+				Trade {
+					pool: PoolType::Stableswap(102),
+					asset_in: 102,
+					asset_out: base,
+				},
+			];
+			assert_ok!(Router::force_insert_route(
+				RuntimeOrigin::root(),
+				AssetPair {
+					asset_in: asset,
+					asset_out: base
+				},
+				BoundedVec::truncate_from(route)
+			));
+
+			let input = EvmDataWriter::new_with_selector(AggregatorInterface::LatestAnswer).build();
+
+			// kinda weird that asset order is reversed, would expect WBTC/USDT instead
+			let address = chainlink_adapter::encode_oracle_address(base, asset, OraclePeriod::TenMinutes, [0; 8]);
+			let mut handle = MockHandle {
+				input: input.clone(),
+				context: Context {
+					address,
+					caller: Default::default(),
+					apparent_value: Default::default(),
+				},
+				code_address: address,
+				is_static: true,
+			};
+			let PrecompileOutput { output, .. } =
+				ChainlinkOraclePrecompile::<hydradx_runtime::Runtime>::execute(&mut handle).unwrap();
+			let precompile_price = U256::from(output.as_slice());
+
+			let (.., output) = Executor::<Runtime>::view(
+				CallContext {
+					contract: reference_oracle,
+					sender: Default::default(),
+					origin: Default::default(),
+				},
+				input,
+				100_000,
+			);
+			let dia_price = U256::from(output.as_slice());
+
+			//TODO: shouldn't be exactly same, but comparable within 5%
+			assert_eq!(dia_price, precompile_price);
```

Comment:
these numbers should be comparable within 5%, or atleast same order of magnitude :)